### ISSUE TITLE
Switch indent issue with short array notation

### ISSUE THREAD

**Issue by odinuv** [squizlabs/PHP_CodeSniffer#786]
<?php
The code:

```
switch ($something) {
    case 'value1':
        $do;
        return ["message" => $message];
    case 'value2':
        $do;
        $do;
        return ["message" => $message];
    default:
        "dosomething";
}
```

Produces the following errors in PHPCS 2.4 (used to pass in 2.3.x):

```
D:\wwwroot\utils>phpcs --standard=psr2 testcs.php

FILE: D:\wwwroot\utils\testcs.php
----------------------------------------------------------------------
FOUND 5 ERRORS AFFECTING 5 LINES
----------------------------------------------------------------------
  7 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     4
  8 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
  9 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
 11 | ERROR | [x] Line indented incorrectly; expected 8 spaces, found
    |       |     4
 12 | ERROR | [x] Line indented incorrectly; expected at least 12
    |       |     spaces, found 8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 5 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 51ms; Memory: 3Mb
```

While this code passes without errors:
```
<?php

switch ($something) {
    case 'value1':
        $do;
        return array('message' => $message);
    case 'value2':
        $do;
        $do;
        return array('message' => $message);
    default:
        "dosomething";
}
```

**Comment by derrabus** [comment]
I ran into the same problem today. Since the `phpcs` run is part of our CI, we had to roll back to 2.3 because of this regression.

Test file:
```php
<?php

namespace Acme;

class Foo
{
    public function bar($x)
    {
        switch ($x) {
            case 1:
                return [
                   'Hello',
                   'World'
                ];
            case 2:
                return [
                    'Another',
                    'Test'
                ];
            default:
                throw new \InvalidArgumentException;
        }
    }
}
```

Output:
```
----------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
 15 | ERROR | [x] Line indented incorrectly; expected 16 spaces,
    |       |     found 12
 20 | ERROR | [x] Line indented incorrectly; expected 16 spaces,
    |       |     found 12
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------
```

So, CodeSniffer wants me to indent the second `case` and the `default` stements, which feels very wrong. Note that it does not complain about the first `case`.

**Comment by derrabus** [comment]
The regression seems to be introduced with commit be47642ca134907b3fa3f941f4cc5d002d34043d (ping @gsherwood).

**Comment by dereuromark** [comment]
Yeah, we also got this regression/bug now in our repo ( https://travis-ci.org/cakephp/chronos/builds/92927002 ).

**Comment by montchr** [comment]
I'm having a similar issue, but it's expecting 12 and finds 16 for nested arrays:

```php
register_taxonomy( 'pedestal_source', [ 'pedestal_link' ], [
    'hierarchical'      => true,
    'public'            => true,
    'show_in_nav_menus' => true,
    'show_ui'           => true,
    'show_admin_column' => true,
    'query_var'         => true,
    'rewrite'           => true,
    'capabilities'      => [
        'manage_terms'  => 'edit_posts',
        'edit_terms'    => 'edit_posts',
        'delete_terms'  => 'edit_posts',
        'assign_terms'  => 'edit_posts',
    ],
] );
```

The items within the `capabilities` nested array will trigger the error “Multi-line function call not indented correctly; expected 12 spaces but found 16”

**Comment by gsherwood** [comment]
@montchr I can't replicate your issue, but it's not the same thing as the main one. Please submit a new report with steps to replicate (what standard, PHPCS version, command you are running etc).

**Comment by gsherwood** [comment]
@derrabus 100% correct about the commit that caused the bug. I've added an additional check for this specific case (scope closer being used to return a short array) and that fixes all the bits of sample code that have been posted around the place.

**Comment by derrabus** [comment]
I've tested with the master branch and cannot reproduce this issue anymore. Thank you for the quick fix. :-)

**Comment by gsherwood** [comment]
Thanks a lot for testing it for me.

**Comment by dereuromark** [comment]
Will you release a bugfix version?
Right now all CS builds are still failing for the projects, of course.

**Comment by gsherwood** [comment]
> Will you release a bugfix version?

I will release 2.5.0 soon. You can always revert to 2.3.x in the meantime.
